utils/data_utils.py

```diff
@@ -16,130 +16,146 @@
+        self.data = from_uri(self.data_tiled_uri, api_key=self.data_tiled_api_key, timeout=httpx.Timeout(30.0))
+
+
+    def get_data_project_names(self):
+        """
+        Get available project names from the main Tiled container,
+        filtered by types that can be processed (Container and ArrayClient)
+        """
+        project_names = [
+            project
+            for project in list(self.data)
+            if isinstance(self.data[project], (Container, ArrayClient))
+        ]
+        return project_names
+
+
+    def get_data_sequence_by_name(self, project_name):
+        """
+        Data sequences may be given directly inside the main client container,
+        but can also be additionally encapsulated in a folder, multiple container or in a .nxs file.
+        We make use of specs to figure out the path to the 3d data.
+        """
+        project_client = self.data[project_name]
+        # If the project directly points to an array, directly return it
+        if isinstance(project_client, ArrayClient):
+            return project_client
+        # If project_name points to a container
+        elif isinstance(project_client, Container):
+            # Check if the specs give us information about which sub-container to access
+            specs = project_client.specs
+            if any(spec.name == "NXtomoproc" for spec in specs):
+                # Example for how to access data if the project container corresponds to a
+                # nexus-file following the NXtomoproc definition
+                # TODO: This assumes that a validator has checked the file on ingestion
+                # Otherwise we should first test if the path holds data
+                return project_client["entry/data/data"]
+            # Enter the container and return first element
+            # if it represents an array
+            if len(list(project_client)) == 1:
+                sequence_client = project_client.values()[0]
+                if isinstance(sequence_client, ArrayClient):
+                    return sequence_client
+        return None
+
+
+    def get_data_shape_by_name(self, project_name):
+        """
+        Retrieve shape of the data
+        """
+        project_container = self.get_data_sequence_by_name(project_name)
+        if project_container:
+            return project_container.shape
+        return None
+
+    
+    @staticmethod
+    def get_annotated_segmented_results(json_file_path="exported_annotation_data.json"):
+        annotated_slices = []
+        with open(json_file_path, "r") as f:
+            for line in f:
+                if line.strip():
+                    json_data = json.loads(line)
+                    json_data["data"] = json.loads(json_data["data"])
+                    annotated_slices = list(json_data["data"][0]["annotations"].keys())
+        return annotated_slices
+
+    
+    @staticmethod
+    def DEV_load_exported_json_data(file_path, USER_NAME, PROJECT_NAME):
```

Comment:
What's the need for the `DEV` name?

Reply:
Most functions that were implemented as a placeholder while some of the connections to Tiled/Computing API/... were missing were marked with a `DEV_` prefix.